packages/fxa-settings/src/lib/oauth/hooks.tsx

```diff
@@ -179,7 +179,7 @@ export function tryAgainError() {
  */
 export function useFinishOAuthFlowHandler(
   authClient: AuthClient,
-  integration: Integration
+  integration: SigninOAuthIntegration
```

Comment:
This may also be a `SignupOAuthIntegration` or one from reset password. Do we need to change this?

Reply:
Ah good call. Seems like only a few things are actually need off integration and quite a few different integrations could be sent in. To make it more concise I've switched to an inline pick on the Integration type with minimum required set of fields.